youtube_dlc/extractor/niconico.py

```diff
@@ -284,80 +360,132 @@ def _real_extract(self, url):
+                                     expected=True)
+            else:
+                raise ExtractorError('%s reports error: %s' % (self.IE_NAME, error_code))
+
+        # Start extracting video formats
+        formats = []
+
+        # Get HTML5 videos info
+        dmc_info = try_get(api_data, lambda x: x['video']['dmcInfo'])
+
+        quality_info = dmc_info.get('quality')
+        for audio_quality in quality_info.get('audios') or {}:
+            for video_quality in quality_info.get('videos') or {}:
+                if not audio_quality.get('available') or not video_quality.get('available'):
+                    continue
+                formats.append(self._extract_format_for_quality(
+                    api_data, video_id, audio_quality, video_quality))
+
+        # Get flv/swf info
+        video_real_url = try_get(api_data, lambda x: x['video']['smileInfo']['url'])
+        is_economy = video_real_url.endswith('low')
+
+        if is_economy:
+            self.report_warning('Site is currently in economy mode! You will only have access to lower quality streams')
+
+        # Invoking ffprobe to determine resolution
+        pp = FFmpegPostProcessor(self._downloader)
+        cookies = self._get_cookies('https://nicovideo.jp').output(header='', sep='; path=/; domain=nicovideo.jp;\n')
+
+        self.to_screen('%s: %s' % (video_id, 'Checking smile format with ffprobe'))
+
+        metadata = pp.get_metadata_object(video_real_url, ['-cookies', cookies])
+
+        v_stream = a_stream = {}
+
+        # Some complex swf files doesn't have video stream (e.g. nm4809023)
+        for stream in metadata['streams']:
+            if stream['codec_type'] == 'video':
+                v_stream = stream
+            elif stream['codec_type'] == 'audio':
+                a_stream = stream
+
+        # Community restricted videos seem to have issues with the thumb API not returning anything at all
+        filesize = int(
+            (get_video_info_xml('size_high') if not is_economy else get_video_info_xml('size_low'))
+            or metadata['format']['size']
+        )
+
+        extension = (
+            get_video_info_xml('movie_type')
+            or 'mp4' if 'mp4' in metadata['format']['format_name'] else metadata['format']['format_name']
+            or determine_ext(video_real_url)
+        )
+
+        timestamp = (
+            parse_iso8601(get_video_info_web('first_retrieve'))
+            or unified_timestamp(get_video_info_web('postedDateTime'))
+        )
+
+        # According to my personal research,
+        # 'creation_time' tag on video stream of re-encoded SMILEVIDEO mp4 files are '1970-01-01T00:00:00.000000Z'.
+        metadata_timestamp = (
+            parse_iso8601(try_get(v_stream, lambda x: x['tags']['creation_time']))
+            or timestamp
+        )
+
+        # According to compconf and my personal research, smile videos from pre-2017 are always better quality than their DMC counterparts
```

Comment:
```suggestion
        # According to compconf, smile videos from pre-2017 are always better quality than their DMC counterparts
```

or better, give source